modules/appeals_api/lib/appeals_api/token_validation_client.rb

```diff
@@ -42,7 +53,16 @@ def validate_token!(audience:, token:, scopes:)
 
       matching_scope = scopes.find { |scope| permitted.include?(scope) }
 
-      raise ::Common::Exceptions::Forbidden unless matching_scope
+      raise ::Common::Exceptions::Forbidden if matching_scope.blank?
+
+      body = JSON.parse(response.body)
+
+      is_veteran_token = permitted.any? { |s| s.start_with? 'veteran/' }
+
+      TokenValidationResult.new(
+        scopes: permitted,
+        veteran_icn: is_veteran_token ? body.dig('data', 'attributes', 'act', 'icn') : nil
```

Comment:
Interesting background info resulting from [my thread with the Pivot team](https://lighthouseva.slack.com/archives/CTN4N4PL2/p1701738807902909) - below is an example of the type of information we get back from the validation server (which is `body` here). I'm using the `data/attributes/act/icn` path to the ICN within this data instead of the `data/launch/patient` path for a reason - see the comments:
```json
{
  "data": {
    "id": "<id>",
    "type": "validated_token",
    "attributes": {
      "jti": "<jti>",
      "iss": "https://deptva-eval.okta.com/oauth2/auskff1rm4M2lFaIt2p7",
      "aud": "https://sandbox-api.va.gov/services/appeals",
      "iat": 1701736897,
      "exp": 1701740497,
      "cid": "<cid>",
      "uid": "<uid>",
      "scp": ["veteran/AppealsStatus.read"],
      "sub": "c6c1f001-a28f-4f21-8c67-ab0a6a554da0",
      "act": {
        "icn": "1234567890V123456", // ICN of the user that the token represents is here
        "npi": null,
        // Notice that the type is "patient", but it could also be "clinician", although we probably won't see that in our API
        "type": "patient",                     
        "sec_id": null,
        "vista_id": null,
        "first_name": null,
        "last_name": null,
        "middle_name": null
      },
      "launch": {
        // ICN is here too, but again, "patient" may not always be a key here, so in this PR, we're using the ICN from the "act" section above instead.
        "patient": "1234567890V123456"
      },
      "ver": 1
    }
  }
}
```